modules/images/webp-uploads/fallback.js

```diff
@@ -4,21 +4,57 @@ window.wpPerfLab = window.wpPerfLab || {};
 	window.wpPerfLab.webpUploadsFallbackWebpImages = function( media ) {
 		for ( var i = 0; i < media.length; i++ ) {
 			try {
-				if ( ! media[i].media_details.sources || ! media[i].media_details.sources['image/jpeg'] ) {
-					continue;
-				}
-
-				var ext = media[i].media_details.sources['image/jpeg'].file.match( /\.\w+$/i );
-				if ( ! ext || ! ext[0] ) {
-					continue;
-				}
+				var image         = media[ i ],
+					media_details = image.media_details,
+					media_sources = media_details.sources,
+					sizes         = media_details.sizes,
+					sizes_keys    = Object.keys( sizes ),
+					images        = document.querySelectorAll( 'img.wp-image-' + image.id );
 
-				var images = document.querySelectorAll( 'img.wp-image-' + media[i].id );
 				for ( var j = 0; j < images.length; j++ ) {
-					images[j].src = images[j].src.replace( /\.webp$/i, ext[0] );
-					var srcset = images[j].getAttribute( 'srcset' );
+
+					var src    = images[ j ].src,
+						srcset = images[ j ].getAttribute( 'srcset' );
+
+					// If there are no sizes, then replace src through sources, there is nothing more to replace.
+					if ( media_sources ) {
+						src = src.replace( media_sources['image/webp'].file, media_sources['image/jpeg'].file );
+						images[ j ].setAttribute( 'src', src );
+						break;
+					}
+
+					// If the full image has no JPEG version available, no sub-size will have JPEG available either.
+					if ( ! sizes.full.sources['image/jpeg'] ) {
+						continue;
+					}
+
+					for ( var k = 0; k < sizes_keys.length; k++ ) {
+						var media_sizes_sources = sizes[ sizes_keys[ k ] ].sources;
+						if ( ! media_sizes_sources || ! media_sizes_sources['image/webp'] || ! media_sizes_sources['image/jpeg'] ) {
+							continue;
+						}
+
+						// Check to see if the image src has any size set, then update it.
+						if ( media_sizes_sources['image/webp'].source_url === src ) {
+							src = media_sizes_sources['image/jpeg'].source_url;
+
+							// If there is no srcset and the src has been replaced, there is nothing more to replace.
+							if ( ! srcset ) {
+								break;
+							}
+						}
+
+						if ( srcset && media_sizes_sources['image/webp'] ) {
```

Comment:
The one variable is already checked above, so the additional check here is unnecessary:
```suggestion
						if ( srcset ) {
```